frontend/src/app/store/router-store.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { Params, Router } from '@angular/router';
+import { Store } from './types';
+import { Observable, Subject } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+
+/**
+ * Synchronises stored parameters with the route query parameters
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class RouterStore implements Store {
+    params$: Observable<Params>;
+    paramUpdates$ = new Subject<Params>();
+
+    constructor(
+        private router: Router
+    ) {
+        this.params$ = this.router.routerState.root.queryParams;
```

Comment:
Why do we need this Observable, if we already have a Subject + a `currentParams` function? Perhaps `paramUpdate$` as a `BehaviorSubject` with `this.router.routerState.root.queryParams` as initial value would be a more concise way to go?